tuxemon/technique.py

```diff
@@ -297,25 +297,38 @@ def changestats(self, user: Monster, target: Monster) -> EffectResult:
         '''
         statsmaster = [self.statspeed, self.stathp, self.statarmour, self.statmelee, self.statranged, self.statdodge]
         statslugs = ['speed', 'hp', 'armour', 'melee', 'ranged', 'dodge']
+        newstatvalue = 0
         for stat, slug in zip(statsmaster, statslugs):
             if not stat:
                 continue
-            value = stat['value']
-            dividing = stat['dividing']
+            minvalue = stat['minvalue']
+            maxvalue = stat['maxvalue']
+            iterator = stat['iterator']
             override = stat['overridetofull']
             basestatvalue = getattr(target, slug)
-            if value > 0 and override == False:
-                if dividing == False:
-                    newstatvalue = basestatvalue + value
-                else:
+            if maxvalue:
+                value = random.randint(minvalue,maxvalue)
+            else:
+                value = minvalue
+            if value > 0 and  override == False:
+                if iterator == '/':
                     newstatvalue = basestatvalue // value
-                success = True
-                setattr(target, slug, newstatvalue)
+                    setattr(target, slug, newstatvalue)
+                elif iterator == '+':
+                    newstatvalue = basestatvalue + value
+                    setattr(target, slug, newstatvalue)
+                elif iterator == '-':
+                    newstatvalue = basestatvalue - value
+                    setattr(target, slug, newstatvalue)
+                elif iterator == '*':
+                    newstatvalue = basestatvalue * value
+                    setattr(target, slug, newstatvalue)
```

Comment:
This could be done in a more compact way:
```python
ops_dict = {
    "+": operator.add,
    "-": operator.sub,
    "*": operator.mul,
    "/": operator.floordiv,
}

newsatvalue = ops_dict[iterator](basestatvalue, value)
setattr(target, slug, newstatvalue)
```

Reply:
fixed